compilers/concrete-compiler/compiler/lib/Conversion/FHETensorOpsToLinalg/TensorOpsToLinalg.cpp

```diff
@@ -2228,6 +2235,195 @@ struct FancyAssignToSfcForall
   };
 };
 
+/// This rewrite pattern transforms any instance of operators
+/// `arith.index_cast` and `arith.extsi` to an instance of `linalg.generic`.
+///
+/// Example:
+///
+///   %output = arith.index_cast %input : tensor<3xi4> to tensor<3xindex>
+///
+/// becomes:
+///
+///   #map = affine_map<(d0) -> (d0)>
+///
+///   %empty = tensor.empty() : tensor<3xindex>
+///   %output = linalg.generic
```

Comment:
The operation `tensor.elemwise_unary` can probably save some boilerplate code in this pattern.

Reply:
Are we sure it's supported through the pipeline :grin:

Reply:
I understand that it's a pain to add support for an extra op to the pipeline, but I think it's more idiomatic / canonical to use `elemwise_unary`.

Reply:
It seems `elemwise_unary` only supports certain set of operations so we can't do this.